ui/desktop/electron-app/src/session/manager.js

```diff
@@ -0,0 +1,38 @@
+const Session = require('./index.js');
+
+class SessionManager {
+  #sessions;
+
+  constructor() {
+    this.#sessions = [];
+  }
+
+  /**
+   * Start a session and track it.
+   * Returns session proxy details if successfully started.
+   * @param {string} addr
+   * @param {string} target_id
+   * @param {string} token
+   * @param {string} host_id
+   */
+  async start(addr, target_id, token, host_id) {
+    const session = new Session(addr, target_id, token, host_id);
+    this.#sessions.push(session);
+    return session.start();
+  }
+
+  /**
+   * Stop a session using identifier.
+   * @param {string} session_id
+   */
+  async stopById(session_id) {
+    const session = this.#sessions.find((session) => session.id === session_id);
+    return session.stop();
+  }
+
+  stopAll() {
+  }
+}
+
+const sessionManager = new SessionManager();
```

Comment:
I suggest instantiating this outside the class file so that the class itself can be reused if we wanted to.

Reply:
Enforced creation on initial load to ensure there will only ever be one session manager for the app. Would you recommend updating current folder structure to include `models` to contain class files?

Reply:
It may be a singleton, but what if we wanted to unit test the class?  Or what if we wanted to segregate sessions by some arbitrary characteristic using multiple managers?  It is bit more flexible to separate the class from the instantiation even for singletons.

Reply:
I see the point of this. But for learning purposes, where will be a good place to instantiate this? within the `electron-app` or within `desktop`? Just trying to figure out good practices here.

Reply:
I'm noticing that `RuntimeSettings` uses this pattern so let's set this discussion aside for another time and address both cases at once.

Reply:
I took a stab at this by moving `Session` and `SessionManager` class definitions to a new `models` folder. Singleton for session manager now exists under `src/session/manager.js`. Happy to refactor `RuntimeSettings` if this approach makes sense.

Reply:
Looks good.  I wonder if "services" or "utils" is more applicable since something like the manager isn't exactly a model.  It's debatable whether a session is in this case.

Reply:
Ok now I understand this better. I do not have a strong opinion, but I see your point here. That said, I see it more as a service than as util.

Reply:
I'm leaning towards 'services' too. Will update.

Reply:
Moved 'session/manager' to 'services/session-manager' plus additional refactoring of origin runtimesettings into new 'services/origin'.